x/callback/abci.go

```diff
@@ -51,14 +51,32 @@ func callbackExec(ctx sdk.Context, k keeper.Keeper, wk types.WasmKeeperExpected,
 			// todo: throw error event with details on failure. will do in diff PR
 		}
 
-		unusedGas := callbackGasLimit - gasUsed
 		logger.Info(
 			"callback executed with pending gas",
 			"contract_address", callback.ContractAddress,
 			"job_id", callback.GetJobId(),
-			"unused_gas", unusedGas,
+			"used_gas", gasUsed,
 		)
-		// todo: refund any leftover to the address which reserved the callback. will do in diff PR
+
+		// Calculate current tx fees based on gasConsumed. Refund any leftover to the address which reserved the callback
+		txFeesConsumed := k.CalculateTransactionFees(ctx, gasUsed)
+		if txFeesConsumed.IsLT(*callback.FeeSplit.TransactionFees) {
```

Comment:
My understanding is that:
- Consumer commits to paying at least X amount of coins;
- On callback exec if what consumer committed to paying is lower than what he committed, then he'll get a refund.

My question is:
- What'd happen if txFeesConsumed > transactionFeesCommitted?

Reply:
Currently, in case txFeesConsumed > transactionFeesCommitted, the loss is just swallowed.
Figured this was the simplest way for v1.

But open to alternatives.

Reply:
I think it's also fine, tbh. 

Just should be noted somewhere

Reply:
Yea will mention in ADR